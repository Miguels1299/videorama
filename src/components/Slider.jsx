import React from 'react'
import 'netslider/dist/styles.min.css';
import SliderContainer from './SliderContainer';
import NetSlider from 'netslider';
import PropTypes from 'prop-types';

function SliderTemplate(props){
    return (
        <div className="slider-container-wrapper">
            <SliderContainer videoModel={props.videoModel} model={props.model}> </SliderContainer>
        </div>
    )
}

export const Slider = ({titulo,peliculas},props) => {
    return (
            <div className="netslider-container" style={{ overflow:'hidden', height:'300px' }} >
                <h1 style={{textAlign:'left', paddingLeft:'4%', marginTop:"40px"}}>{titulo}</h1>
                <NetSlider className="netslider_title_card" data={peliculas} slideTemplate={props => (<SliderTemplate {...props} />)}/>
            </div>
    )
}

Slider.propTypes = {
    titulo: PropTypes.string,
    peliculas: PropTypes.array,
}
