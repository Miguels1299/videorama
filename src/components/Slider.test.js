import React from 'react';
import {shallow} from 'enzyme';
import {Slider} from './Slider';

describe('Pruebas en <Slider />', () => {
    const titulo = "Pelicula";      
    const peliculas = [
        {
            id: 1,
            image:
                "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABZkJbs3ZqZsg0PrW4wd4VqEDOJljJdlSqeFthGwNQ49mX2UqkUwQBwquT3In6z7DtSCT1z8TRwYY_uKprg5a_rcx8v947ocQj2RyjkQ_dhwCqwDaS1O9wWYmqOiv.jpg",
            imageHighRes:
                "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABZkJbs3ZqZsg0PrW4wd4VqEDOJljJdlSqeFthGwNQ49mX2UqkUwQBwquT3In6z7DtSCT1z8TRwYY_uKprg5a_rcx8v947ocQj2RyjkQ_dhwCqwDaS1O9wWYmqOiv.jpg",
        },
        {
            id: 2,
            image:
                "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABRGA5vLR9JW1REzbP5nniTuHbCNr3dr08nHGABq59Qrn3gzf3bNn0uk86pffebgVDwQr1ikfI7s7upNkiQZ4XR9VwAo.jpg",
            imageHighRes:
                "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABRGA5vLR9JW1REzbP5nniTuHbCNr3dr08nHGABq59Qrn3gzf3bNn0uk86pffebgVDwQr1ikfI7s7upNkiQZ4XR9VwAo.jpg",
        }
    ];  
    const wrapper = shallow( <Slider titulo={titulo} peliculas={peliculas} /> );
   

    test ('Mostrar componente correctamente',() => {
        expect(wrapper).toMatchSnapshot();
    });

    test ('Debe tener un titulo', () => {
        const h1 = wrapper.find('h1');
        expect( h1.text().trim()).toBe(titulo);
    });

    test ('Debe tener peliculas', () => {
        const t = wrapper.find('t');
        expect(t.prop('data')).toBe(peliculas);
    });

    test('Debe tener clases netslider-container', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('netslider-container')).toBe(true);
    })
    
})
