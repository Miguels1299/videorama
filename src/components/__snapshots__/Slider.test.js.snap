// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Slider /> Mostrar componente correctamente 1`] = `
<div
  className="netslider-container"
  style={
    Object {
      "height": "300px",
      "overflow": "hidden",
    }
  }
>
  <h1
    style={
      Object {
        "marginTop": "40px",
        "paddingLeft": "4%",
        "textAlign": "left",
      }
    }
  >
    Pelicula
  </h1>
  <t
    className="netslider_title_card"
    data={
      Array [
        Object {
          "id": 1,
          "image": "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABZkJbs3ZqZsg0PrW4wd4VqEDOJljJdlSqeFthGwNQ49mX2UqkUwQBwquT3In6z7DtSCT1z8TRwYY_uKprg5a_rcx8v947ocQj2RyjkQ_dhwCqwDaS1O9wWYmqOiv.jpg",
          "imageHighRes": "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABZkJbs3ZqZsg0PrW4wd4VqEDOJljJdlSqeFthGwNQ49mX2UqkUwQBwquT3In6z7DtSCT1z8TRwYY_uKprg5a_rcx8v947ocQj2RyjkQ_dhwCqwDaS1O9wWYmqOiv.jpg",
        },
        Object {
          "id": 2,
          "image": "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABRGA5vLR9JW1REzbP5nniTuHbCNr3dr08nHGABq59Qrn3gzf3bNn0uk86pffebgVDwQr1ikfI7s7upNkiQZ4XR9VwAo.jpg",
          "imageHighRes": "https://occ-0-1229-116.1.nflxso.net/dnm/api/v6/X194eJsgWBDE2aQbaNdmCXGUP-Y/AAAABRGA5vLR9JW1REzbP5nniTuHbCNr3dr08nHGABq59Qrn3gzf3bNn0uk86pffebgVDwQr1ikfI7s7upNkiQZ4XR9VwAo.jpg",
        },
      ]
    }
    hideRowHeader={true}
    id="83"
    jawBoneRankNum={null}
    slideTemplate={[Function]}
    videoRoot="reference"
  />
</div>
`;
